### PR TITLE
refactor: remove os_errmsg and os_msg functions

### PR DESCRIPTION
**Pull request by dundargoc**
Instead replace them with fprintf and printf.

### CODE REVIEW

File: src/nvim/lua/executor.c [hunk 1]
```diff
@@ -254,8 +254,8 @@ static int nlua_luv_thread_common_cfpcall(lua_State *lstate, int nargs, int nres
     if (status == LUA_ERRMEM && !(flags & LUVF_CALLBACK_NOEXIT)) {
       // Terminate this thread, as the main thread may be able to continue
       // execution.
-      os_errmsg(e_outofmem);
-      os_errmsg("\n");
+      fprintf(stderr, "%s", e_outofmem);
```
Comment (bfredl) [inline src/nvim/lua/executor.c:257]: we can clean up all callsites with static strings like these, just `fprintf(stderr, e_outofmem)` would work. Although right here we might want `fprintf(stderr, "%s\n",  e_outofmem)` .  (imagine being able to actually use printf when writing to stdio lol)

File: src/nvim/main.c [hunk 2]
```diff
@@ -2188,38 +2189,39 @@ static void usage(void)
...
-  os_msg(_("  --api-info            Write msgpack-encoded API metadata to stdout\n"));
-  os_msg(_("  --clean               \"Factory defaults\" (skip user config and plugins, shada)\n"));
-  os_msg(_("  --embed               Use stdin/stdout as a msgpack-rpc channel\n"));
-  os_msg(_("  --headless            Don't start a user interface\n"));
-  os_msg(_("  --listen <address>    Serve RPC API from this address\n"));
-  os_msg(_("  --remote[-subcommand] Execute commands remotely on a server\n"));
-  os_msg(_("  --server <address>    Specify RPC server to send commands to\n"));
-  os_msg(_("  --startuptime <file>  Write startup timing messages to <file>\n"));
-  os_msg(_("\nSee \":help startup-options\" for all options.\n"));
+  printf("%s", _("Usage:\n"));
```
Comment (bfredl) [inline src/nvim/main.c:2192]: also in cases like this. gettext formatstring is used anyway in a lot of other places.

File: src/nvim/main.c [hunk 3]
```diff
@@ -1624,9 +1619,8 @@ static void open_script_files(mparm_T *parmp)
   if (parmp->scriptout) {
     scriptout = os_fopen(parmp->scriptout, parmp->scriptout_append ? APPENDBIN : WRITEBIN);
     if (scriptout == NULL) {
-      os_errmsg(_("Cannot open for script output: \""));
-      os_errmsg(parmp->scriptout);
-      os_errmsg("\"\n");
+      fprintf(stderr, _("Cannot open for script output: \""));
```
Comment (zeertzjq) [inline src/nvim/main.c:1622]: This message is translated, so I think it's better to make it a format argument.
```suggestion
      fprintf(stderr, "%s", _("Cannot open for script output: \""));
```

File: src/nvim/lua/executor.c [hunk 4]
```diff
@@ -815,12 +813,12 @@ void nlua_init(char **argv, int argc, int lua_arg0)
 
   lua_State *lstate = luaL_newstate();
   if (lstate == NULL) {
-    os_errmsg(_("E970: Failed to initialize lua interpreter\n"));
+    fprintf(stderr, _("E970: Failed to initialize lua interpreter\n"));
```
Comment (zeertzjq) [inline src/nvim/lua/executor.c:816]: ditto
```suggestion
    fprintf(stderr, "%s", _("E970: Failed to initialize lua interpreter\n"));
```

File: src/nvim/lua/executor.c [hunk 5]
```diff
@@ -815,12 +813,12 @@ void nlua_init(char **argv, int argc, int lua_arg0)
...
     os_exit(1);
   }
   luaL_openlibs(lstate);
   if (!nlua_state_init(lstate)) {
-    os_errmsg(_("E970: Failed to initialize builtin lua modules\n"));
+    fprintf(stderr, _("E970: Failed to initialize builtin lua modules\n"));
```
Comment (zeertzjq) [inline src/nvim/lua/executor.c:821]: ditto
```suggestion
    fprintf(stderr, "%s", _("E970: Failed to initialize builtin lua modules\n"));
```

File: src/nvim/main.c [hunk 6]
```diff
@@ -2158,25 +2152,21 @@ static void print_mainerr(const char *errstr, const char *str)
...
   if (str != NULL) {
-    os_errmsg(": \"");
-    os_errmsg(str);
-    os_errmsg("\"");
+    fprintf(stderr, ": \"%s\"", str);
   }
-  os_errmsg(_("\nMore info with \""));
-  os_errmsg(prgname);
-  os_errmsg(" -h\"\n");
+  fprintf(stderr, _("\nMore info with \""));
```
Comment (zeertzjq) [inline src/nvim/main.c:2160]: ditto
```suggestion
  fprintf(stderr, "%s", _("\nMore info with \""));
```

File: src/nvim/main.c [hunk 7]
```diff
@@ -2187,38 +2177,38 @@ static void usage(void)
...
+  printf(_("  --                    Only file names after this\n"));
+  printf(_("  --api-info            Write msgpack-encoded API metadata to stdout\n"));
+  printf(_("  --clean               \"Factory defaults\" (skip user config and plugins, shada)\n"));
+  printf(_("  --embed               Use stdin/stdout as a msgpack-rpc channel\n"));
+  printf(_("  --headless            Don't start a user interface\n"));
+  printf(_("  --listen <address>    Serve RPC API from this address\n"));
+  printf(_("  --remote[-subcommand] Execute commands remotely on a server\n"));
+  printf(_("  --server <address>    Specify RPC server to send commands to\n"));
+  printf(_("  --startuptime <file>  Write startup timing messages to <file>\n"));
+  printf(_("\nSee \":help startup-options\" for all options.\n"));
```
Comment (zeertzjq) [inline src/nvim/main.c:2211]: ditto
```suggestion
  printf("%s", _("Usage:\n"));
  printf("%s", _("  nvim [options] [file ...]\n"));
  printf("%s", _("\nOptions:\n"));
  printf("%s", _("  --cmd <cmd>           Execute <cmd> before any config\n"));
  printf("%s", _("  +<cmd>, -c <cmd>      Execute <cmd> after config and first file\n"));
  printf("%s", _("  -l <script> [args...] Execute Lua <script> (with optional args)\n"));
  printf("%s", _("  -S <session>          Source <session> after loading the first file\n"));
  printf("%s", _("  -s <scriptin>         Read Normal mode commands from <scriptin>\n"));
  printf("%s", _("  -u <config>           Use this config file\n"));
  printf("\n");
  printf("%s", _("  -d                    Diff mode\n"));
  printf("%s", _("  -es, -Es              Silent (batch) mode\n"));
  printf("%s", _("  -h, --help            Print this help message\n"));
  printf("%s", _("  -i <shada>            Use this shada file\n"));
  printf("%s", _("  -n                    No swap file, use memory only\n"));
  printf("%s", _("  -o[N]                 Open N windows (default: one per file)\n"));
  printf("%s", _("  -O[N]                 Open N vertical windows (default: one per file)\n"));
  printf("%s", _("  -p[N]                 Open N tab pages (default: one per file)\n"));
  printf("%s", _("  -R                    Read-only (view) mode\n"));
  printf("%s", _("  -v, --version         Print version information\n"));
  printf("%s", _("  -V[N][file]           Verbose [level][file]\n"));
  printf("\n");
  printf("%s", _("  --                    Only file names after this\n"));
  printf("%s", _("  --api-info            Write msgpack-encoded API metadata to stdout\n"));
  printf("%s", _("  --clean               \"Factory defaults\" (skip user config and plugins, shada)\n"));
  printf("%s", _("  --embed               Use stdin/stdout as a msgpack-rpc channel\n"));
  printf("%s", _("  --headless            Don't start a user interface\n"));
  printf("%s", _("  --listen <address>    Serve RPC API from this address\n"));
  printf("%s", _("  --remote[-subcommand] Execute commands remotely on a server\n"));
  printf("%s", _("  --server <address>    Specify RPC server to send commands to\n"));
  printf("%s", _("  --startuptime <file>  Write startup timing messages to <file>\n"));
  printf("%s", _("\nSee \":help startup-options\" for all options.\n"));
```

File: src/nvim/message.c [hunk 8]
```diff
@@ -225,7 +225,7 @@ int verb_msg(const char *s)
 }
 
 /// Displays the string 's' on the status line
-/// When terminal not initialized (yet) os_errmsg(..) is used.
+/// When terminal not initialized (yet) fprintf(..) is used.
```
Comment (zeertzjq) [inline src/nvim/message.c:228]:
```suggestion
/// When terminal not initialized (yet) printf("%s", ..) is used.
```

File: src/nvim/message.c [hunk 9]
```diff
@@ -748,7 +748,7 @@ bool emsg_multiline(const char *s, bool multiline)
 /// emsg() - display an error message
 ///
 /// Rings the bell, if appropriate, and calls message() to do the real work
-/// When terminal not initialized (yet) os_errmsg(..) is used.
+/// When terminal not initialized (yet) fprintf(..) is used.
```
Comment (zeertzjq) [inline src/nvim/message.c:751]:
```suggestion
/// When terminal not initialized (yet) fprintf(stderr, "%s", ..) is used.
```

File: src/nvim/ex_cmds.c [hunk 10]
```diff
@@ -1486,7 +1486,7 @@ void print_line(linenr_T lnum, int use_number, int list)
 
   msg_start();
   silent_mode = false;
-  info_message = true;  // use os_msg(), not os_errmsg()
+  info_message = true;  // use printf(), not fprintf()
```
Comment (zeertzjq) [inline src/nvim/ex_cmds.c:1489]:
```suggestion
  info_message = true;  // use stdout, not stderr
```

File: src/nvim/main.c [hunk 11]
```diff
@@ -2158,25 +2152,21 @@ static void print_mainerr(const char *errstr, const char *str)
...
+  fprintf(stderr, "%s -h\"\n", prgname);
 }
 
 /// Prints version information for "nvim -v" or "nvim --version".
 static void version(void)
 {
   // TODO(bfred): not like this?
   nlua_init(NULL, 0, -1);
-  info_message = true;  // use os_msg(), not os_errmsg()
+  info_message = true;  // use printf(), not fprintf()
```
Comment (zeertzjq) [inline src/nvim/main.c:2169]:
```suggestion
  info_message = true;  // use stdout, not stderr
```

File: src/nvim/option.c [hunk 12]
```diff
@@ -1487,10 +1487,10 @@ int do_set(char *arg, int opt_flags)
   if (silent_mode && did_show) {
     // After displaying option values in silent mode.
     silent_mode = false;
-    info_message = true;        // use os_msg(), not os_errmsg()
+    info_message = true;        // use printf(), not fprintf()
```
Comment (zeertzjq) [inline src/nvim/option.c:1490]:
```suggestion
    info_message = true;        // use stdout, not stderr
```

File: src/nvim/option.c [hunk 13]
```diff
@@ -1487,10 +1487,10 @@ int do_set(char *arg, int opt_flags)
...
     msg_putchar('\n');
     silent_mode = true;
-    info_message = false;       // use os_msg(), not os_errmsg()
+    info_message = false;       // use printf(), not fprintf()
```
Comment (zeertzjq) [inline src/nvim/option.c:1493]:
```suggestion
    info_message = false;       // use stdout, not stderr
```

File: src/nvim/option.c [hunk 14]
```diff
@@ -3927,7 +3927,7 @@ static void showoneopt(vimoption_T *opt, int opt_flags)
   int save_silent = silent_mode;
 
   silent_mode = false;
-  info_message = true;          // use os_msg(), not os_errmsg()
+  info_message = true;          // use printf(), not fprintf()
```
Comment (zeertzjq) [inline src/nvim/option.c:3930]:
```suggestion
  info_message = true;          // use stdout, not stderr
```

File: src/nvim/main.c [hunk 15]
```diff
@@ -2158,25 +2151,21 @@ static void print_mainerr(const char *errstr, const char *str)
 
   signal_stop();              // kill us with CTRL-C here, if you like
 
-  os_errmsg(prgname);
-  os_errmsg(": ");
-  os_errmsg(_(errstr));
+  fprintf(stderr, "%s: ", prgname);
+  fprintf(stderr, "%s", _(errstr));
```
Comment (zeertzjq) [inline src/nvim/main.c:2155]:
```suggestion
  fprintf(stderr, "%s: %s", prgname, _(errstr));
```

**Review by bfredl — APPROVED** [review]

**Pull request merged**